styled/colors.go

```diff
@@ -0,0 +1,107 @@
+package styled
+
+import "fmt"
+
+type Color uint
```

Comment:
Type safety and memory compactness aside, it seems to be much more straightforward if we just use `string` to represent colors and translate them to SGR codes when printing.

It will also be easier to extend more complex color representations (e.g. hex `#123456`) if we just use `string`.